testsuite/MDAnalysisTests/transformations/test_fit.py

```diff
@@ -99,7 +99,7 @@ def test_fit_translation_no_masses(fit_universe):
     # create a universe without masses
     ref_u  = make_Universe()
     # what happens Universe without masses is given?
-    with pytest.raises(AttributeError): 
+    with pytest.raises(ValueError): 
         fit_translation(test_u, ref_u, weights="mass")(ts)
```

Comment:
IMO the resulting error you get here: 
https://github.com/MDAnalysis/mdanalysis/blob/a78307f0d99b7310707ef015cab2908ce4770a4e/package/MDAnalysis/transformations/fit.py#L102-L107

doesn't make sense for the input, since you do give `weights="mass"`. Probably the easiest solution is to just not catch the `TypeError`, since `get_weights` already has good error messages for that one:

https://github.com/MDAnalysis/mdanalysis/blob/a78307f0d99b7310707ef015cab2908ce4770a4e/package/MDAnalysis/lib/util.py#L1355-L1375

And then it's worth checking the error message in this test to make sure you're getting the *right* TypeError. e.g.

```python
with pytest.raises(TypeError) as exc:
   fit_translation(....)
assert 'atoms.masses is missing' in str(exc.value)
```

Reply:
@lilyminium I'm not getting this one clearly, please help me out with this.

Reply:
It would help if you let me know what you don't understand, so I know where to start.

Reply:
```>python
with pytest.raises(TypeError) as exc:
        fit_translation(test_u, ref_u, weights="mass")(ts)
        assert 'atoms.masses is missing' in str(exc.value)
```
@lilyminium Will adding the above finction to `test_fit.py` do the job?

Reply:
You probably want the last line (starting with ``assert``) outside with ``with`` context, but that will be a good test for the changes you make to the code. You should also make changes to the code.

Reply:
What type of code would it be sir?

Reply:
> What type of code would it be

IMO this highlighted `fit_translation` test should not raise a `ValueError` with the error message `"weights must be {'mass', None} or an iterable of the same size as the atomgroup."`, as this error message is not applicable when your test does indeed pass `weights="mass"` into the function. Instead, the problem is that your atoms don't have masses. 

The reason that the said ValueError is raised, is because the `fit_translation` function catches `ValueError`s and `TypeError`s raised by the `get_weights` function. If you look at the `get_weights` function, it raises a TypeError when the atoms don't have masses. This TypeError already has a perfectly good error message. I am suggesting that instead of raising the `ValueError` with the above error message after catching this `TypeError`, you just let the error go through by *not* catching it with `except`.

> sir?

Lily or @lilyminium is fine :-)